EssentialCSharp.Web/Areas/Identity/Pages/Account/Manage/DownloadPersonalData.cshtml.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#nullable disable
+
+using System.Text.Json;
+using EssentialCSharp.Web.Areas.Identity.Data;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace EssentialCSharp.Web.Areas.Identity.Pages.Account.Manage
+{
+    public class DownloadPersonalDataModel : PageModel
+    {
+        private readonly UserManager<EssentialCSharpWebUser> _UserManager;
+        private readonly ILogger<DownloadPersonalDataModel> _Logger;
+
+        public DownloadPersonalDataModel(
+            UserManager<EssentialCSharpWebUser> userManager,
+            ILogger<DownloadPersonalDataModel> logger)
+        {
+            _UserManager = userManager;
+            _Logger = logger;
+        }
+
+        public IActionResult OnGet()
+        {
+            return NotFound();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            EssentialCSharpWebUser user = await _UserManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_UserManager.GetUserId(User)}'.");
+            }
+
+            _Logger.LogInformation("User with ID '{UserId}' asked for their personal data.", _UserManager.GetUserId(User));
+
+            // Only include personal data for download
```

Comment:
Don't we want to include all data?

Reply:
I'm not sure what other data we would include at this point? But this is just grabbing data that is tagged with the "PersonalData" Property Attribute, which just seems the most straightforward at this point. 

Is there something else I should add right now to this? Or just something to keep in mind in the future?